### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by hienphan0111**
### Interface

- [ ] Create a class `Nameable`.

- [ ] Implement a method called `correct_name` 

###  Turn  `Person` class to Nameable

- [ ] Make sure that `Person` class inherits from `Nameable`

- [ ] Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

### Prepare base Decorator

- [ ] Make sure that it inherits from Nameable.

- [ ] In the constructor assign a nameable object from params to an instance variable.

- [ ] Implement the correct_name method that returns the result of the correct_name method of the @nameable.

### Prepare CapitalizeDecorator and TrimmerDecorator

1. For the CapitalizeDecorator:

- [ ] Create a class that inherits from the base Decorator class.

- [ ] Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.

2. For the TrimmerDecorator:

- [ ] Create a class that inherits from the base Decorator class.

- [ ] Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name()
+    puts 'NotImplementedError'
+  end
```
Comment (julie-ify) [inline nameable.rb]: - [ ] This method should raise an exception, not print one.

  ```
  def correct_name
    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
  end
  ```

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name
```
Comment (julie-ify) [inline decorator.rb]: - [ ] Your method does not capitalize the name. Kindly fix that.

  ```
  @nameable.correct_name.capitalize
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @hienphan0111,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @hienphan0111 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Use the Decorator design pattern ✔️
Create class that inherits from the base Decorator class ✔️

## Optional suggestions
[optional] Please kindly ensure that all the empty headers (**key features and setup**) in your README file are filled up appropriately, this will make your README more engaging and informative. 👍 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**